cobbler/utils/threading.py

```diff
@@ -0,0 +1,97 @@
+"""
+This module is responsible for managing the custom common threading logic Cobbler has.
+"""
+
+import logging
+import pathlib
+import time
+from threading import Thread
+
+from cobbler import enums
+from cobbler import utils
+
+
+class CobblerThread(Thread):
+    """
+    This is a custom thread that has a custom logger as well as logic to execute Cobbler triggers.
+    """
+
+    def __init__(self, event_id: str, remote, options: dict, task_name: str, api):
+        """
+        This constructor creates a Cobbler thread which then may be run by calling ``run()``.
+
+        :param event_id: The event-id which is associated with this thread. Also used as thread name
+        :param remote: The Cobbler remote object to execute actions with.
+        :param options: Additional options which can be passed into the Thread.
+        :param task_name: The high level task name which is used to trigger pre- and post-task triggers
+        :param api: The Cobbler api object to resolve information with.
+        """
+        Thread.__init__(self, name=event_id)
```

Comment:
Is there a reason not to use `super()`?

Reply:
Nope. That's a leftover from the old times. Will change that.